test/python/transpiler/test_inverse_cancellation.py

```diff
@@ -368,6 +370,15 @@ def test_parameterized_self_inverse_not_equal_parameter(self):
         new_circ = inverse_pass(qc)
         self.assertEqual(new_circ, qc)
 
+    def test_parameterized_self_inverse_not_equal_parameter_2(self):
+        """Test that a parameterized self inverse gate doesn't cancel incorrectly.
+        This test, checks two gates with the same name but different parameters."""
+        qc = QuantumCircuit(1)
+        qc.rz(0, 0)
+        qc.rz(3.14159, 0)
+        inverse_pass = InverseCancellation([RZGate(0)])
+        new_circ = inverse_pass(qc)
```

Comment:
We should add an assertion here that `new_circ` is the same as the input as there was nothing to simplify:

```suggestion
        new_circ = inverse_pass(qc)
        self.assertEqual(qc, new_circ)
```

Reply:
oops :)))